source/kraken/apply_disruption.cpp

```diff
@@ -208,6 +208,10 @@ struct add_impacts_visitor : public apply_impacts_visitor {
                                                          meta.production_date);
             if (! r && ! mvj->get_base_vj().empty()) {
                 r = mvj->get_base_vj().at(0)->route;
+            } else {
```

Comment:
This should be a `else if (! r)`.
You may reorganize the code, but you get the spirit :wink:

Reply:
I let you change that when you merged the work. This is your pluging place :hear_no_evil: